### PR TITLE
Valgrind: leak supression updates.; leak check on every commit; small/medium sized leak fixes.

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR:
- Enables leak checking on CI.
- Updates the leak suppression with some redis suppressions
- **Important** Includes leak fix for Series DUMP/RESTORE due to rdb loading of series not freeing the default chunk (meaning a leak of the default chunk size per key. See #609
 - Includes some small leak fixes that were detected in the course of this PR.

### CODE REVIEW

File: src/compaction.c [hunk 1]
```diff
@@ -310,7 +310,9 @@ void MaxMinReadContext(void *contextPtr, RedisModuleIO *io) {
     size_t len = 1;
     context->maxValue = RedisModule_LoadDouble(io);
     context->minValue = RedisModule_LoadDouble(io);
-    context->isResetted = RedisModule_LoadStringBuffer(io, &len)[0];
```
Comment (filipecosta90) [inline src/compaction.c:313]: @danni-m  small leak 1

File: src/tsdb.c [hunk 2]
```diff
@@ -514,6 +514,7 @@ int SeriesCreateRulesFromGlobalConfig(RedisModuleCtx *ctx,
         if (RedisModule_KeyType(compactedKey) != REDISMODULE_KEYTYPE_EMPTY) {
             // TODO: should we break here? Is log enough?
             RM_LOG_WARNING(ctx, "Cannot create compacted key, key '%s' already exists", destKey);
+            RedisModule_FreeString(ctx, destKey);
```
Comment (filipecosta90) [inline src/tsdb.c:517]: @danni-m  small leak 2

File: tests/flow/test_ts_madd.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def test_partial_madd():
 
         now = int(time.time() * 1000)
         res = r.execute_command("ts.madd", 'test_key1', "*", 10, 'test_key2', 2000, 20, 'test_key3', 3000, 30)
-        assert now <= res[0] and now + 2 >= res[0]
```
Comment (filipecosta90) [inline tests/flow/test_ts_madd.py:68]: @danni-m  the second condition might not be true if within valgrind takes more than 2 seconds to execute ts.madd. Removed this condition given it was flaky

File: tests/redis_valgrind.sup [hunk 4]
```diff
@@ -15,3 +15,203 @@
...
+   <invalid_write_size_8_compression_appendInteger>
+   Memcheck:Addr8
+   fun:appendBits
+   fun:appendInteger
+   fun:Compressed_Append
+   fun:Compressed_AddSample
+}
+
+{
+   <temporary_supression_redistimeseries_UNCOMPRESSED_due_to_untrimmed_series_DUMP_LOAD>
```
Comment (filipecosta90) [inline tests/redis_valgrind.sup:78]: @danni-m @gkorland 
temporarily includes supression of Series DUMP/RESTORE due to untrimmed series with empty chunks being stored as is ( meaning we can have chunks that are empty saved ).

File: src/tsdb.c (hunk 2 above)
Comment (danni-m) [reply src/tsdb.c:517]: @filipecosta90 im not sure this is a leak, the destKey is kept in inside the `CompactionRule` struct via `SeriesAddRule`

File: src/Makefile [hunk 5]
```diff
@@ -200,7 +200,6 @@ tests: $(TARGET)
 ifeq ($(COV),1)
 	$(COVERAGE_RESET)
 endif
-	$(COVERAGE_RESET)
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply src/Makefile:203]: this line was added y mistake. We don't need to reset coverage outside coverage tests.

File: src/tsdb.c (hunk 2 above)
Comment (filipecosta90) [reply src/tsdb.c:517]: @danni-m if you uncomment the fix line and run ``. You will receive the leak report as follows: `make test SIMPLE=1 VGD=1 TEST=test_globalconfigs.py`

```
353304:signal-handler (1608728599) Received SIGTERM scheduling shutdown...
353304:M 23 Dec 2020 13:03:19.868 # User requested shutdown...
353304:M 23 Dec 2020 13:03:19.875 # Redis is now ready to exit, bye bye...
==353304== 254 (80 direct, 174 indirect) bytes in 5 blocks are definitely lost in loss record 782 of 862
==353304==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==353304==    by 0x152030: zmalloc (zmalloc.c:99)
==353304==    by 0x16415D: createObject (object.c:42)
==353304==    by 0x1DC6D4: RM_CreateStringPrintf (module.c:1050)
==353304==    by 0x5DCD520: SeriesCreateRulesFromGlobalConfig (tsdb.c:510)
==353304==    by 0x5DC5D9E: add (module.c:833)
==353304==    by 0x5DC606C: TSDB_add (module.c:876)
==353304==    by 0x1DB8C4: RedisModuleCommandDispatcher (module.c:643)
==353304==    by 0x1499BC: call (server.c:3354)
==353304==    by 0x14AC46: processCommand (server.c:3783)
==353304==    by 0x15DA71: processCommandAndResetClient (networking.c:1857)
==353304==    by 0x15DCDE: processInputBuffer (networking.c:1940)
```

File: tests/redis_valgrind.sup (hunk 4 above)
Comment (filipecosta90) [reply tests/redis_valgrind.sup:78]: see #609

**Review by rafie — APPROVED** [review]

File: src/tsdb.c (hunk 2 above)
Comment (danni-m) [reply src/tsdb.c:517]: Looking at the code again you are right.

**Review by danni-m — DISMISSED** [review]

File: src/chunk.c [hunk 6]
```diff
@@ -220,7 +221,8 @@ void Uncompressed_LoadFromRDB(Chunk_t **chunk, struct RedisModuleIO *io) {
     uncompchunk->base_timestamp = RedisModule_LoadUnsigned(io);
     uncompchunk->num_samples = RedisModule_LoadUnsigned(io);
     uncompchunk->size = RedisModule_LoadUnsigned(io);
-
-    uncompchunk->samples = (Sample *)RedisModule_LoadStringBuffer(io, NULL);
+    size_t string_buffer_size;
+    uncompchunk->samples = (Sample *)RedisModule_LoadStringBuffer(io, &string_buffer_size);
+    assert(uncompchunk->size == string_buffer_size);
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply src/chunk.c]: No failures. Was just ensuring all was ok during the changes. Removed it :)

**Review by danni-m — APPROVED** [review]

**Pull request merged**